### PR TITLE
workflow: add retention period

### PR DESCRIPTION
**Pull request by sevein**
In this commit, watchers learn a new configuration attribute `retentionPeriod`
passed to the workflow as part of the blob event struct.

In `enduro.toml`, set up the retention period as follows:

```
[[watchers.foobar]]
retentionPeriod = "24h"
```

The format follows Go's convention to express duration:
https://golang.org/pkg/time/#ParseDuration.

The zero-value, e.g. when the retention period config is missing or zero, will
cause the original to not be deleted.

The implementation is based on Cadence workflow timers. The timer is launched
as long as the package processing has completed successfully. The deletion
occurs when the timer future is released.

### CODE REVIEW

File: enduro.toml [hunk 1]
```diff
@@ -24,12 +24,14 @@ key = "36J9X8EZI4KEV1G7EHXA"
...
 region = "us-west-1"
 pipeline = "am"
+retentionPeriod = "10s"
 
 [[watcher.filesystem]]
 name = "dev-fs"
 path = "./hack/watched-dir"
 inotify = true
 pipeline = "am"
+retentionPeriod = "11s"
```
Comment (replaceafill) [inline enduro.toml]: Is there any reason this watcher needs more retention time than the minio watcher?

File: internal/workflow/processing.go [hunk 2]
```diff
@@ -491,38 +512,9 @@ func (a *CleanUpActivity) Execute(ctx context.Context, tinfo *TransferInfo) erro
...
-	}
-
-	tinfo.CollectionID = col.ID
-
-	return tinfo, nil
-}
-
-func updatePackageStatusLocalActivity(ctx context.Context, colsvc collection.Service, tinfo *TransferInfo) error {
-	info := activity.GetInfo(ctx)
+	a.manager.Watcher.Watchers()
```
Comment (replaceafill) [inline internal/workflow/processing.go]: What's the purpose of this call?

Comment (sevein) [reply internal/workflow/processing.go]: Ops, good catch. Deleted.

File: enduro.toml (hunk 1 above)
Comment (sevein) [reply enduro.toml]: Just for testing purposes, so you can tell the actual value expected applies.

**Pull request merged**